### PR TITLE
deflakes test-delete-service and test-killed-instances

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- deflakes test-delete-service and test-killed-instances

## Why are we making these changes?

We would like all our tests to run and catch regressions.

### CODE REVIEW

File: waiter/integration/waiter/basic_test.clj [hunk 1]
```diff
@@ -344,62 +345,56 @@
...
-; ...
-; "marathonclj.rest.apps$delete_app.invoke(apps.clj:17)",
-; "waiter.scheduler.marathon.MarathonScheduler$fn__18236.invoke(marathon.clj:260)",
-; "waiter.scheduler$retry_on_transient_server_exceptions_fn$fn__17306.invoke(scheduler.clj:91)"
-; ...
-; "waiter.scheduler.marathon.MarathonScheduler.delete_app(marathon.clj:257)",
-; "waiter.core$delete_app_handler.invokeStatic(core.clj:410)"
-; ...
-(deftest ^:parallel ^:integration-fast ^:explicit test-delete-app
+(deftest ^:parallel ^:integration-fast ^:dev test-delete-service
```
Comment (dposada) [inline waiter/integration/waiter/basic_test.clj]: Why `^dev`?

File: waiter/integration/waiter/basic_test.clj [hunk 2]
```diff
@@ -468,43 +463,38 @@
       (is (str/includes? set-cookie "HttpOnly=true"))
       (is (= (System/getProperty "user.name") (str body))))))
 
-; Marked explicit due to:
-;   FAIL in (test-killed-instances)
-;   Not intersection between used and killed instances!
-;   expected: (not-empty (set/intersection all-instance-ids killed-instance-ids))
-;     actual: (not (not-empty #{}))
-(deftest ^:parallel ^:integration-slow ^:explicit test-killed-instances
+(deftest ^:parallel ^:integration-slow ^:dev test-killed-instances
```
Comment (dposada) [inline waiter/integration/waiter/basic_test.clj]: Why `^dev`?

File: waiter/integration/waiter/basic_test.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/integration/waiter/basic_test.clj]: Removed the `:dev`, I was using the tag to run only these tests on my parent branch that fixes all the explicit tests.

File: waiter/integration/waiter/basic_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/integration/waiter/basic_test.clj]: Removed the `:dev`, I was using the tag to run only these tests on my parent branch that fixes all the explicit tests.

**Comment by dposada** [comment]
Will merge once the build is green

**Comment by shamsimam** [comment]
Build is green.

**Pull request merged**